api/client/formatter/network.go

```diff
@@ -0,0 +1,120 @@
+package formatter
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/engine-api/types"
+)
+
+const (
+	defaultNetworkTableFormat = "table {{.ID}}\t{{.Name}}\t{{.Driver}}"
+
+	networkIDHeader = "NETWORK ID"
+	ipv6Header      = "IPV6"
+	internalHeader  = "INTERNAL"
+	// scopeHeader = "SCOPE"
+)
+
+// NetworkContext contains network specific information required by the formatter,
+// encapsulate a Context struct.
+type NetworkContext struct {
+	Context
+	// Networks
+	Networks []types.NetworkResource
+}
+
+func (ctx NetworkContext) Write() {
+	switch ctx.Format {
+	case tableFormatKey:
+		if ctx.Quiet {
+			ctx.Format = defaultQuietFormat
+		} else {
+			ctx.Format = defaultNetworkTableFormat
+		}
+	case rawFormatKey:
+		if ctx.Quiet {
+			ctx.Format = `network_id: {{.ID}}`
+		} else {
+			ctx.Format = `network_id: {{.ID}}\nname: {{.Name}}\ndriver: {{.Driver}}\n`
+		}
+	}
+
+	ctx.buffer = bytes.NewBufferString("")
+	ctx.preformat()
+
+	tmpl, err := ctx.parseFormat()
+	if err != nil {
+		return
+	}
+
+	for _, network := range ctx.Networks {
+		networkCtx := &networkContext{
+			trunc: ctx.Trunc,
+			n:     network,
+		}
+		err = ctx.contextFormat(tmpl, networkCtx)
+		if err != nil {
+			return
+		}
+	}
+
+	ctx.postformat(tmpl, &networkContext{})
+}
+
+type networkContext struct {
+	baseSubContext
+	trunc bool
+	n     types.NetworkResource
+}
+
+func (c *networkContext) ID() string {
+	c.addHeader(networkIDHeader)
+	if c.trunc {
+		return stringid.TruncateID(c.n.ID)
+	}
+	return c.n.ID
+}
+
+func (c *networkContext) Name() string {
+	c.addHeader(nameHeader)
+	return c.n.Name
+}
+
+func (c *networkContext) Driver() string {
+	c.addHeader(driverHeader)
+	return c.n.Driver
+}
+
+// YES/NO vs true/false vs 1/0 ?
```

Comment:
@thaJeztah I wasn't sure what to put here for boolean values. How do we usually print them ?